src/dwarf/Gparser.c

```diff
@@ -574,6 +584,8 @@ get_rs_cache (unw_addr_space_t as, intrmask_t *saved_maskp)
   if ((atomic_read (&as->cache_generation) != atomic_read (&cache->generation))
        || !cache->hash)
     {
+      /* cache_size is only set in the global_cache, copy it over before flushing */
+      cache->log_size = as->global_cache.log_size;
```

Comment:
Thats most likely true, I did not dive into the 'set cache size' thing much yet, though I don't have an idea how to handle it.